src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -528,30 +575,75 @@ def getMSOutputTransferInfo(self, wflow):
             wflow['TransferDone'] = True
         return wflow
 
+    def setParentDatasets(self, wflow):
+        """
+        Used to resolve parent datasets for a workflow.
+        :param  wflow:   A MSRuleCleaner workflow representation
+        :return:         The workflow object
+        """
+        if wflow['InputDataset'] and wflow['IncludeParents']:
+            childDataset = wflow['InputDataset']
+            parentDataset = findParent([childDataset], self.msConfig['dbsUrl'])
```

Comment:
Todor, haven't we discussed that this API will return:
```
parentDataset = {"pileup_dset_name": None}
```
in case of errors?

In such case, the if statement below would fail to catch those errors and would actually append `None` to a list of parent datasets.

Reply:
@amaltaro I need to understand, what are the conditions under which the following line would be hit [1], and what is the difference from the other error here [2], so I can see if I should treat them separately. 

[1]
https://github.com/dmwm/WMCore/blob/e049d6cf05e49e99843226874a6587c7c760adf0/src/python/WMCore/MicroService/Unified/Common.py#L646

[2]
https://github.com/dmwm/WMCore/blob/e049d6cf05e49e99843226874a6587c7c760adf0/src/python/WMCore/MicroService/Unified/Common.py#L654

Reply:
I believe a DBS/CMSWEB unavailable would fall under the case [1]. Case isn't clear to me, I guess only if the data object isn't iterable or something like that.

Reply:
Done